tests/devices/system_tests/test_oav_to_redis_system.py

```diff
@@ -0,0 +1,51 @@
+import asyncio
+from unittest.mock import AsyncMock, patch
+
+import pytest
+from aiohttp.client_exceptions import ClientConnectorError
+from ophyd_async.core import DeviceCollector
+
+from dodal.devices.oav.oav_to_redis_forwarder import OAVToRedisForwarder
+
+
+@pytest.fixture
+@patch("dodal.devices.oav.oav_to_redis_forwarder.StrictRedis")
+def oav_to_redis_forwarder(_, RE):
+    with DeviceCollector():
+        oav_forwarder = OAVToRedisForwarder("BL04I", "", "")
+    oav_forwarder.redis_client.hset = AsyncMock()
+    return oav_forwarder
+
+
+@pytest.mark.s03  # Doesn't actually depend on s03, just on DLS network. See https://github.com/DiamondLightSource/mx-bluesky/issues/183
+async def test_given_stream_url_is_not_a_real_webpage_when_kickoff_then_error(
+    oav_to_redis_forwarder: OAVToRedisForwarder,
+):
+    oav_to_redis_forwarder.stream_url = AsyncMock()
+    oav_to_redis_forwarder.stream_url.get_value.return_value = (
+        "http://www.this_is_not_a_valid_webpage.com/"
+    )
+    with pytest.raises(ClientConnectorError):
+        await oav_to_redis_forwarder.kickoff()
+
+
+@pytest.mark.s03  # Doesn't actually depend on s03, just on DLS network. See https://github.com/DiamondLightSource/mx-bluesky/issues/183
+async def test_given_stream_url_is_real_webpage_but_not_mjpg_when_kickoff_then_error(
```

Comment:
This test isn't passing for me device the PV's haven't been connected. Using a mock device with the `DeviceCollector` will fix it